### ISSUE TITLE
How are the 3D Gaussians, which are not rendered from the current viewpoint, updated through the loss gradients?

### ISSUE THREAD

**Issue by democheng** [graphdeco-inria/gaussian-splatting#712]
Thank you for your gread work and the open source code.
Before ask this question, I have read all the issues related with "grad", but I  didn't find the answer.

I loaded a model with 1200 images, and I always choose the same image when selecting the training viewpoint; when I set the gradients of the 3D Gaussians seen by that image to False, the loss still changes.
```
    def visible_training(self, idxs_input):
        # no_grad_idxs = ~idxs_input # set False to the unvisible 3D Gaussians
        no_grad_idxs = idxs_input # set False to the visible 3D Gaussians, the loss should remain unchanged, In fact, it has changed. 
        for idx in range(len(self.optimizer.param_groups)):
            self.optimizer.param_groups[idx]['params'][0][no_grad_idxs].requires_grad_(False)
        self.optimizer.step()
        self.optimizer.zero_grad(set_to_none = True)
        for idx in range(len(self.optimizer.param_groups)):
            self.optimizer.param_groups[idx]['params'][0][no_grad_idxs].requires_grad_(True)
        return

```

```
idxs_input = render_pkg["visibility_filter"]
```

I want retrain part of the large scene with some image list, but the 3D Gaussians, which are not rendered from the current viewpoint, are also updated through the loss gradients.
